src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js

```diff
@@ -65,20 +69,23 @@ foam.CLASS({
         user.setTwoFactorSecret(key);
         userDAO.put_(x, user);
 
-        if ( ! generateQrCode ) {
-          return key;
-        }
-
         try {
           EmailConfig service = (EmailConfig) x.get("emailConfig");
           String name = service == null ? "FOAM" : service.getDisplayName();
           String path = String.format("/%s:%s", name, user.getEmail());
           String query = String.format("secret=%s&issuer=%s&algorithm=%s", key, name, getAlgorithm());
           URI uri = new URI("otpauth", "totp", path, query, null);
-          return "data:image/svg+xml;charset=UTF-8," + QrCode.encodeText(uri.toASCIIString(), QrCode.Ecc.MEDIUM).toSvgString(0);
+
+          otpKey.setKey(key);
+          otpKey.setQrCode("data:image/svg+xml;charset=UTF-8,"
+            + QrCode.encodeText(uri.toASCIIString(), QrCode.Ecc.MEDIUM).toSvgString(0));
+          return otpKey;
         } catch ( Throwable t ) {
-          throw new RuntimeException(t);
+          logger.error("Error when generating QR code: " + t);
```

Comment:
Should we be swallowing this error like this or still throwing an exception? To me, something exceptional has happened if we've failed to retrieve a QR code and if I'm requesting an OTPKey, I'll never know that something went wrong.